libraries/bsg_manycore_config.cpp

```diff
@@ -135,13 +135,28 @@ int hb_mc_config_init(const hb_mc_config_raw_t raw[HB_MC_CONFIG_MAX],
         config->vcache_block_words  = raw[HB_MC_CONFIG_VCACHE_BLOCK_WORDS];
 
         idx = raw[HB_MC_CONFIG_VCACHE_STRIPE_WORDS];
-        if (idx < config->vcache_block_words) { 
+        if (idx < config->vcache_block_words) {
                 bsg_pr_err("%s: Invalid vcache stripe size %d: stripe size "
                            "cannot be smaller than vcache block size %d: %s\n",
                            __func__, idx, config->vcache_block_words, error_init_help);
                 return HB_MC_INVALID;
         }
         config->vcache_stripe_words = idx;
 
+        idx = raw[HB_MC_CONFIG_IO_REMOTE_LOAD_CAP];
+        if ((idx > HB_MC_CONFIG_MAX_REMOTE_LOADS)){
+                bsg_pr_err("%s: Invalid remote load caps %d: %s\n",
+                           __func__, idx, error_init_help);
+                return HB_MC_INVALID;
+        }
+        config->io_remote_load_cap = idx;
+
+
+        /* the flowing flow control parameters are checked at test_rom*/
+
+        config->io_endpoint_max_out_credits = raw[HB_MC_CONFIG_IO_EP_MAX_OUT_CREDITS];
+        config->io_host_credits_cap = raw[HB_MC_CONFIG_IO_HOST_CREDITS_CAP];
+
```

Comment:
yeah, but test_rom isn't run every time. You should check that they're sensible values here -- i.e non zero, not 0xFFFFFFFF, and less than something else.